saleor/graphql/schema.graphql

```diff
@@ -1717,6 +1717,17 @@ type ProductDelete {
   product: Product
 }
 
+input ProductFilterInput {
+  name: String
+  name__icontains: String
+  category: ID
+  product_type: ID
+  price: Float
+  price__lt: Float
```

Comment:
Currently, we have a few filters that we've added manually to the `products` query and I think we should move them to the `filters` object so we don't duplicate them and have a single place to actually perform filtering. These are:
- `priceLte` & `priceGte` - these could be deprecated, as we already have them in in `filters`
- `collections` - allows you to get products that belong to given collections - I think it shouldn't be a problem to move it to have that handled by `ProductFilter`,
- `attributes` - this one is tricky because it uses very specific format, but if you go back in the Github history you'd find an implementation of that filter with django-filters.
- `stockAvailability` - should be moved to `ProductFilter`
- `query` - this one is used in a few more resolvers and I know that our Dashboard 2.0 uses it for searching objects. I think that I'd rather drop `name` and `name__icontains` for now in favor of `query`. Question is if we can move `query` to the `filters` object easily.

Reply:
Do we have our own solutions for deprecation argument/input fields? 
If I understand it correctly, this feature is not supported by graphene.
https://github.com/graphql-python/graphene/issues/799
https://github.com/graphql/graphql-spec/issues/197

Reply:
Right, I guess we cannot deprecate an argument in any way. It won't look very elegant but I think  we should keep the old ones for now and try to implement as many as we can within the `filters` object. Once frontend apps migrate to the new approach we'll drop the old ones.

Reply:
@korycins I have a few more thoughts about this particular filter, after discussion with @patrys. It doesn't make much sense to have the exact `price` filter - usually, you'd want to filter by prices ranges. I think that we could replace all three price filter with something like:
```
type PriceFilter {
  min: Float
  max: Float
}

input ProductFilterInput {
  price: PriceFilter
}
```

If we do that, I guess we won't need to use any of "__lt", "__gt" or "__icontains" suffixes in our filter names. That in turns means that maybe we could use camel case for filter fields names.

Reply:
@maarcingebala Sounds good.